lib/index.js

```diff
@@ -34,12 +34,13 @@ function response (httpResponse, data) {
     },
     json () {
       return new Promise((resolve, reject) => {
-        const err = MOPointer.alloc().initWithValue(null)
-        const obj = NSJSONSerialization.JSONObjectWithData_options_error(data, 0, err)
-        if (obj) {
+        const str = NSString.alloc().initWithData_encoding(data, NSUTF8StringEncoding)
+        if (str) {
+          // parse errors are turned into exceptions, which cause promise to be rejected
+          const obj = JSON.parse(str)
           resolve(obj)
         } else {
-          reject(err.value())
+          reject(new Error("Could not parse JSON because it is not valid UTF-8 data."))
```

Comment:
Throwing in the `executor` is the same as `reject(err)` as far as I can tell:

> If an error is thrown in the executor function, the promise is rejected. The return value of the executor is ignored.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise